lib/utils/analytics/index.js

```diff
@@ -29,147 +28,148 @@ const logError = (type, error) => {
-    }
-    if (areAnalyticsDisabled && !isForced) return null;
-    if (!cacheDirPath) return null;
-    const limit = pLimit(3);
-    return fse.readdir(cacheDirPath).then(
-      (dirFilenames) => {
-        if (!options.serverlessExecutionSpan) process.nextTick(markServerlessRunEnd);
-        return BbPromise.all(
-          dirFilenames.map((dirFilename) =>
-            limit(() => {
-              if (serverlessRunEndTime) return null;
-              if (!isUuid(dirFilename)) return null;
-              return fse.readJson(join(cacheDirPath, dirFilename)).then(
-                (data) => {
-                  if (data && data.payload) {
-                    const timestamp = Number(data.timestamp);
-                    if (timestamp > timestampWeekBefore) {
-                      if (!analyticsUrl) return null;
-                      return request(data.payload, {
-                        id: dirFilename,
-                        timeout: 3000,
-                      });
-                    }
-                  } else {
-                    logError(`Invalid cached data ${dirFilename}`, data);
-                  }
-                  // Invalid or stale event, do not send, and remove from cache
-                  return fse.unlink(join(cacheDirPath, dirFilename)).catch((error) => {
-                    logError(`Could not remove cache file ${dirFilename}`, error);
-                  });
-                },
-                (readJsonError) => {
-                  if (readJsonError.code === 'ENOENT') return null; // Race condition
-                  logError(`Cannot read cache file: ${dirFilename}`, readJsonError);
-                  return fse.unlink(join(cacheDirPath, dirFilename)).catch((error) => {
-                    logError(`Could not remove cache file ${dirFilename}`, error);
-                  });
-                }
-              );
-            })
-          )
-        );
-      },
-      (readdirError) => {
-        if (readdirError.code !== 'ENOENT') logError('Cannot access cache dir', readdirError);
-      }
+            try {
+              return await fse.unlink(join(cacheDirPath, dirFilename));
+            } catch (error) {
+              logError(`Could not remove cache file ${dirFilename}`, error);
+            }
+          } catch (readJsonError) {
+            if (readJsonError.code === 'ENOENT') return null; // Race condition
+            logError(`Cannot read cache file: ${dirFilename}`, readJsonError);
+            try {
+              return await fse.unlink(join(cacheDirPath, dirFilename));
+            } catch (error) {
+              logError(`Could not remove cache file ${dirFilename}`, error);
+            }
+          }
+          return null;
+        })
+      )
     );
-  }).then(() => null); // Do not leak any result
+  } catch (readdirError) {
+    if (readdirError.code !== 'ENOENT') logError('Cannot access cache dir', readdirError);
+  }
+
+  return null;
```

Comment:
This `return` originally was intended to supress array result from `Promise.all` and here it's moved, and no longer applied in same logic point.